cvat/apps/custom_annotation/views.py

```diff
@@ -0,0 +1,302 @@
+
+# Copyright (C) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from django.http import HttpResponse, JsonResponse, HttpResponseBadRequest, QueryDict
+from django.core.exceptions import ObjectDoesNotExist
+from django.shortcuts import render
+from rules.contrib.views import permission_required, objectgetter
+from cvat.apps.authentication.decorators import login_required
+from cvat.apps.engine.models import Task as TaskModel
+from cvat.apps.engine import annotation, task
+
+import django_rq
+import subprocess
+import fnmatch
+import logging
+import json
+import os
+import rq
+
+from cvat.apps.engine.log import slogger
+
+from .model_loader import ModelLoader, read_model_config, get_blob_props, get_model_label_map
+from .image_loader import ImageLoader
+import os.path as osp
+from os import walk
+import json
+import fnmatch
+
+def get_image_data(path_to_data):
+    def get_image_key(item):
+        return int(osp.splitext(osp.basename(item))[0])
+
+    image_list = []
+    for root, _, filenames in walk(path_to_data):
+        for filename in fnmatch.filter(filenames, '*.jpg'):
+                image_list.append(osp.join(root, filename))
+
+    image_list.sort(key=get_image_key)
+    return ImageLoader(image_list)
+
+def load_model(model_file, weights_file, config_file):
+    config = read_model_config(config_file)
+    model = (model_file, weights_file)
+
+    blob_params = get_blob_props(config)
+    class_names = get_model_label_map(config)
+    model =  ModelLoader(path_to_model=model, blob_params=blob_params)
+    model.load()
+
+    return model, class_names
+
+def process_detections(detections, path_to_conv_script):
+    import importlib.util
+    spec = importlib.util.spec_from_file_location('converter', path_to_conv_script)
+    converter = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(converter)
```

Comment:
Is it save to run the line? Do we need a special "sandbox" solution to run custom code? (https://stackoverflow.com/questions/3068139/how-can-i-sandbox-python-in-pure-python)

Reply:
Just look at available options. In trivial case we can use `exec somePythonCode in {'__builtins__': {}}, {}` together with 'admin' right to add a new model.